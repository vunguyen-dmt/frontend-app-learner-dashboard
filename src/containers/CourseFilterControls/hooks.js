import React from 'react';
import { useToggle } from '@edx/paragon';

import { StrictDict } from 'utils';
import track from 'tracking';

import * as module from './hooks';

export const state = StrictDict({
  target: (val) => React.useState(val), // eslint-disable-line
});

export const useCourseFilterControlsData = ({
  filters,
  setFilters,
  setSortBy,
}) => {
  const [isOpen, toggleOpen, toggleClose] = useToggle(false);
  const [target, setTarget] = module.state.target(null);

  const handleFilterChange = ({ target: { checked, value } }) => {
    const update = checked ? setFilters.add : setFilters.remove;
    update(value);
  };
  const handleSortChange = ({ target: { value } }) => {
    setSortBy(value);
  };

  const open = () => {
    track.filter.filterClicked();
    toggleOpen();
  };

  const close = () => {
    track.filter.filterOptionSelected(filters);
    toggleClose();
  };

  return {
    isOpen,
    open,
    close,
    target,
    setTarget,
    handleFilterChange,
    handleSortChange,
  };
};

export default useCourseFilterControlsData;
