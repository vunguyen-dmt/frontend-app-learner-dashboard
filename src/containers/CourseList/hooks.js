import React from 'react';

import { useCheckboxSetValues, useWindowSize, breakpoints } from '@edx/paragon';
import queryString from 'query-string';

import { ListPageSize, SortKeys, FilterKeys } from 'data/constants/app';
import { reduxHooks } from 'hooks';
import { StrictDict } from 'utils';

import * as module from './hooks';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth;
};

export const state = StrictDict({
  sortBy: (val) => React.useState(val), // eslint-disable-line
});

export const useCourseListData = () => {
  const [filters, setFilters] = useCheckboxSetValues([FilterKeys.inProgress]);
  const [sortBy, setSortBy] = module.state.sortBy(SortKeys.enrolled);
  const pageNumber = reduxHooks.usePageNumber();
  const querySearch = queryString.parse(window.location.search, { parseNumbers: true });

  const { numPages, visible } = reduxHooks.useCurrentCourseList({
    sortBy,
    filters,
    pageSize: querySearch?.disable_pagination === 1 ? 0 : ListPageSize,
  });

  const handleRemoveFilter = (filter) => () => setFilters.remove(filter);
  const setPageNumber = reduxHooks.useSetPageNumber();

  return {
    pageNumber,
    numPages,
    setPageNumber,
    visibleList: visible,
    filterOptions: {
      sortBy,
      setSortBy,
      filters,
      setFilters,
      handleRemoveFilter,
    },
    showFilters: filters.length > 0,
  };
};

export default useCourseListData;
