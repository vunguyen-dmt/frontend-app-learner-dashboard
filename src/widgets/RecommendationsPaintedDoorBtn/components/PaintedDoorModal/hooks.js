import React from 'react';
import { StrictDict } from 'utils';

import * as module from './hooks';

export const state = StrictDict({
  isModalOpen: (val) => React.useState(val), // eslint-disable-line
});

export const usePaintedDoorModal = () => {
  const [isModalOpen, setIsModalOpen] = module.state.isModalOpen(false);
  const toggleModal = () => setIsModalOpen(!isModalOpen);

  return {
    isModalOpen,
    toggleModal,
  };
};
