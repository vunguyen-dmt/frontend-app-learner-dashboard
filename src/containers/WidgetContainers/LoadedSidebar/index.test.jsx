import { shallow } from 'enzyme';

import hooks from 'widgets/ProductRecommendations/hooks';
import { mockFooterRecommendationsHook } from 'widgets/ProductRecommendations/testData';
import WidgetSidebar from '.';

jest.mock('widgets/LookingForChallengeWidget', () => 'LookingForChallengeWidget');
jest.mock('widgets/ProductRecommendations/hooks', () => ({
  useShowRecommendationsFooter: jest.fn(),
}));

describe('WidgetSidebar', () => {
  beforeEach(() => jest.resetAllMocks());
  const props = {
    setSidebarShowing: jest.fn(),
  };

  describe('snapshots', () => {
    test('default', () => {
      hooks.useShowRecommendationsFooter.mockReturnValueOnce(
        mockFooterRecommendationsHook.activeControl,
      );
      const wrapper = shallow(<WidgetSidebar {...props} />);

      expect(props.setSidebarShowing).toHaveBeenCalledWith(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test('is hidden when the has the default values', () => {
    hooks.useShowRecommendationsFooter.mockReturnValueOnce(
      mockFooterRecommendationsHook.default,
    );
    const wrapper = shallow(<WidgetSidebar {...props} />);

    expect(props.setSidebarShowing).not.toHaveBeenCalled();
    expect(wrapper.type()).toBeNull();
  });

  test('is hidden when the has the treatment values', () => {
    hooks.useShowRecommendationsFooter.mockReturnValueOnce(
      mockFooterRecommendationsHook.activeTreatment,
    );
    const wrapper = shallow(<WidgetSidebar {...props} />);

    expect(props.setSidebarShowing).not.toHaveBeenCalled();
    expect(wrapper.type()).toBeNull();
  });
});
