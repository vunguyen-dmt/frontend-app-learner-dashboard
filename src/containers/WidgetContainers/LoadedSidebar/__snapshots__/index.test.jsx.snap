// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetSidebar snapshots default 1`] = `
<div
  className="widget-sidebar"
>
  <div
    className="d-flex flex-column"
  >
    <RecommendationsPanel />
  </div>
</div>
`;
